src/main/java/ch/njol/skript/expressions/ExprName.java

```diff
@@ -1,432 +1,442 @@
-	}
-
-	@Override
-	@Nullable
-	public String convert(Object o) {
-		return type.get(o instanceof Slot ? ((Slot) o).getItem() : o);
-	}
-
-	private int changeType = 0;
-
-	// TODO find a better method for handling changes (in general)
-	// e.g. a Changer that takes an object and returns another which should then be saved if applicable (the Changer includes the ChangeMode)
-	@Override
-	@Nullable
-	public Class<?>[] acceptChange(ChangeMode mode) {
-		if (mode == ChangeMode.DELETE && (type.acceptChange & ~PLAYER) != 0 || mode == ChangeMode.RESET)
-			return new Class[0];
-		if (mode != ChangeMode.SET)
-			return null;
-		if ((type.acceptChange & PLAYER) != 0 && Player.class.isAssignableFrom(getExpr().getReturnType())) {
-			changeType = PLAYER;
-		} else if ((type.acceptChange & INVENTORY) != 0 && Inventory.class.isAssignableFrom(getExpr().getReturnType())) {
-			changeType = INVENTORY;
-		} else if ((type.acceptChange & ITEM) != 0 && (getExpr().isSingle() && ChangerUtils.acceptsChange(getExpr(), ChangeMode.SET, ItemStack.class, ItemType.class) || Slot.class.isAssignableFrom(getExpr().getReturnType()))) {
-			changeType = ITEM;
-		} else if ((type.acceptChange & ENTITY) != 0 && Entity.class.isAssignableFrom(getExpr().getReturnType())) {
-			if (type == NameType.NAME && Player.class.isAssignableFrom(getExpr().getReturnType())) {
-				Skript.error("Can't change the Minecraft name of a player. Change the 'display name' or 'tab list name' instead.");
-				return null;
-			}
-			changeType = ENTITY;
-		}
-		return changeType == 0 ? null : CollectionUtils.array(String.class);
-	}
-
-	@Override
-	public void change(Event e, @Nullable Object[] delta, ChangeMode mode) {
-		String name = delta == null ? null : (String) delta[0];
-		if (changeType == ITEM) {
-			if (Slot.class.isAssignableFrom(getExpr().getReturnType())) {
-				for (Slot s : (Slot[]) getExpr().getArray(e)) {
-					ItemStack i = s.getItem();
-					type.set(i, name);
-					s.setItem(i);
-				}
-			} else {
-				Object i = getExpr().getSingle(e);
-				if (i instanceof ItemType) {
-					type.set(i, name);
-					getExpr().change(e, new ItemType[] {(ItemType) i}, ChangeMode.SET);
-					return;
-				}
-
-				if (!(i instanceof ItemStack) && !(i instanceof Slot))
-					return;
-				ItemStack is = i instanceof Slot ? ((Slot) i).getItem() : (ItemStack) i;
-				type.set(is, name);
-				if (i instanceof Slot)
-					((Slot) i).setItem(is);
-				else if (ChangerUtils.acceptsChange(getExpr(), ChangeMode.SET, ItemStack.class))
-					getExpr().change(e, new Object[] {i}, ChangeMode.SET);
-				else
-					getExpr().change(e, new ItemType[] {new ItemType((ItemStack) i)}, ChangeMode.SET);
-			}
-		} else {
-			for (Object o : getExpr().getArray(e)) {
-				if (o instanceof Entity || o instanceof Inventory)
-					type.set(o, name);
-			}
-		}
-	}
-
-	@Override
-	public Class<String> getReturnType() {
-		return String.class;
-	}
-
-	@Override
-	protected String getPropertyName() {
-		return type.name;
-	}
-
-}
+/**
```

Comment:
You will need to change your line endings in your IDE for this class, to prevent this kind of commit happening.
Not only does it kill commit history, but it also prevents us from being able to easily see the line changes.